Refactor ccxt data provider

yuriy-arabskyy opened the pull request.

yuriy-arabskyy committed:
refactor ccxt data provider

bogdanKaftanatiy commented on src/qubx/connectors/ccxt/connection_manager.py:
```diff
@@ -0,0 +1,284 @@
+        subscriber: Callable[[], Awaitable[None]],
+        exchange: Exchange,
+        channel: CtrlChannel,
+        stream_name: str,
```
in most places we use stream name as the main unique identificator of active ws connection. But what I noticed, that is possible to have several active ws connections for the same stream name at the same time. 
For example for some reason we didn't stop old connection, but opened a new one. In such case we are going override connection references
So I think extra validation would be good to add, so we check if old connection for this stream name exists

bogdanKaftanatiy replied:
ah, okay, I see that there is resubscription_cleanup flow added for that, all good

yuriy-arabskyy committed:
implement tests

yuriy-arabskyy committed:
add comprehensive Binance UM swap integration tests

- Fixed orderbook and quote data tests with correct tuple format
- Added comprehensive resubscription scenario tests
- Fixed dynamic instrument addition/removal tests
- Tests all data types: OHLC, trades, orderbook, quotes, funding rates, open interest
- Validates multi-instrument and multi-data-type subscriptions
- Demonstrates real exchange connectivity and data flow
- All tests pass with live Binance UM futures exchange
